sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/KalixServer.scala

```diff
@@ -178,22 +179,38 @@ class KalixServer(applicationContext: ApplicationContext, config: Config) {
   kalixBeanFactory.registerSingleton("viewCreationContext", viewCreationContext)
   kalixBeanFactory.registerSingleton("kalixClient", kalixClient)
 
-  // This little hack allows us to find out which bean is annotated with SpringBootApplication (usually only one).
+  // This little hack allows us to find out which bean is annotated with SpringBootApplication.
   // We need it to find out which packages to scan.
   // Normally, users are expected to have their classes in subpackages of their Main class.
-  val springBootMain =
-    applicationContext.getBeansWithAnnotation(classOf[SpringBootApplication]).values().asScala
+  val mainClass = {
 
-  // TODO: users may define their Kalix components in other packages as well and then use @ComponentScan
-  // to let Spring find them. We should also look for @ComponentScan in the Main class and collect any
-  // scan package declared there
-  val packagesToScan = springBootMain.map(_.getClass.getPackageName).toSet
+    // TODO: we need to verify how reliable is this method of trying to find the main class.
+    //  Would work for any jdk? Should we care?
+    val bootClassOpt =
+      Try(Class.forName(System.getProperty("sun.java.command"))).toOption
+        .filter { cls => cls.getAnnotation(classOf[SpringBootApplication]) != null }
```

Comment:
Here we used to pick the Main class from the application context, but the version that Spring adds to the context is one enhanced by CGLIB and doesn't not have the `@Acl` annotation. 

This workaround to find out the main class may not be JDK dependent so we will need to very this and eventually find a more robust solution.

Reply:
I did some local testing with a few different JDKs and `sun.java.command` seems to be available each time. Although I didn't find a official documentation affirming that (didn't search much neither).

Also, it only works if the pom explicitly declares the main class, which is the case for us. 

If user calls mvn and pass the main class we don't get the same result. 
For example, calling it with:

```java
 mvn compile exec:java -Dexec.mainClass="com.example.demo.DemoApplication"
```

we have as output 

```java
org.codehaus.plexus.classworlds.launcher.Launcher compile exec:java -Dexec.mainClass=com.example.demo.DemoApplication
```

Given that it works with the JDK we use in the container and given that our pom is well configured, we can assume it will work, but if we can find a better solution for this it will be better.